src/components/NavBar/NavBar.tsx

```diff
@@ -362,17 +363,56 @@ export function NavBar(): JSX.Element {
                         <FriendIndicator />
                         <NotificationIndicator onClick={toggleNotifications} />
                         <TurnIndicator />
-                        <span
-                            className="icon-container"
-                            onClick={toggleRightNav}
-                            ref={toggleRightNavButton}
-                        >
+
+                        <span className="icon-container mobile-only" onClick={toggleRightNav}>
                             <PlayerIcon user={user} size={64} />
-                            <span className="username">
-                                {user.username}
-                                <i className="fa fa-caret-down" />
-                            </span>
                         </span>
+
+                        <Menu
+                            title={
+                                <span className="icon-container">
+                                    <PlayerIcon user={user} size={64} />
+                                    <span className="username">{user.username}</span>
+                                </span>
+                            }
+                            to={`/user/view/${user.id}`}
+                            className="profile desktop-only"
+                        >
+                            <Link to={`/user/view/${user.id}`}>
```

Comment:
Given that these are the same contents as desktop (and we don't want them to go out of sync) it might be a good idea to break this into its own component and using it in both places.

Reply:
Yep totally